helm-chart/templates/deployment.yaml

```diff
@@ -38,7 +38,11 @@ spec:
         - configMapRef:
             name: {{ include "json-rpc-relay.fullname" . }}
         - secretRef:
+        {{- if .Values.existingSecret }}
+            name: {{ .Values.existingSecret }}
```

Comment:
It should probably be templated so you can embed the release name in it to support multiple deploys in a single cluster.